ci_framework/roles/openshift_provisioner_node/tasks/main.yml

```diff
@@ -0,0 +1,68 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Create the artifacts directory
+  ansible.builtin.file:
+    path: "{{ cifmw_opn_artifacts_dir }}"
+    state: directory
+  delegate_to: localhost
+
+- name: Add users
+  ansible.builtin.include_tasks:
+    file: add_user.yml
+
+- name: Install required packages for the installer
+  ansible.builtin.include_tasks:
+    file: install_packages.yml
+
+- name: Check network settings are inline with installer requirement
+  ansible.builtin.include_tasks:
+    file: add_bridges.yml
+
+- name: Check required virtualization networks
+  ansible.builtin.include_tasks:
+    file: add_virtual_network.yml
+  loop:
+    - "{{ cifmw_opn_external_bridge_name }}"
+    - "{{ cifmw_opn_prov_bridge_name }}"
+
+- name: Generate random MAC address
+  register: mac_add
+  ansible.builtin.shell:
+    cmd: |
+      hostname | md5sum | sed 's/^\(..\)\(..\)\(..\)\(..\)\(..\).*$/02:\1:\2:\3:\4:\5/'
+
+- name: Cache the MAC addresses
+  ansible.builtin.set_fact:
+    cifmw_opn_bootstrap_boot_mac: "{{ mac_add.stdout }}"
+
+- name: Install and configure DHCP service
+  when:
+    - cifmw_opn_dhcp is defined
+    - cifmw_opn_dhcp | bool
+  vars:
+    mac_address: "{{ cifmw_opn_bootstrap_boot_mac }}"
+    fixed_ipv4: "{{ cifmw_opn_bootstrap_ipv4 }}"
+  ansible.builtin.include_tasks:
+    file: config_dhcpd.yml
+
+- name: Read host SSH fingerprint
+  register: ssh_fp
+  ansible.builtin.command: "cat /etc/ssh/ssh_host_ecdsa_key.pub"
```

Comment:
better to use `ansible.builtin.slurp` module

Reply:
Thanks for the pointer. Will update in the next patch set.